x-pack/plugins/serverless_search/public/layout/nav.tsx

```diff
@@ -52,9 +46,19 @@ const navigationTree: NavigationTreeDefinition = {
             },
             {
               link: 'dashboards',
+              getIsActive: ({ pathNameSerialized, prepend }) => {
+                return pathNameSerialized.startsWith(prepend('/app/dashboards'));
+              },
             },
             {
               link: 'visualize',
+              getIsActive: ({ pathNameSerialized, prepend }) => {
```

Comment:
I added this to maintain the navigation active state when going from the list to the different visualizations detailed view